sources/packages/backend/apps/api/src/services/education-program-offering/education-program-offering-validation.models.ts

```diff
@@ -614,6 +642,188 @@ export class OfferingValidationModel {
       "Not able to find a program related to this offering or it was not provided.",
   })
   programContext: EducationProgramForOfferingValidationContext;
+
+  /**
+   * Institution information required to execute the offering validation.
+   */
+  @IsOptional()
+  // When the context is provided, it must be valid.
+  @IsNotEmptyObject(undefined, {
+    message: "Not able to find the institution of the offering.",
+  })
+  institutionContext?: InstitutionValidationContext;
+
+  /**
+   * Offering mode(s) of online instruction.
+   */
+  @ValidateIf(
+    // Validate when the property must have a value or it already has a value.
+    (offering: OfferingValidationModel) =>
+      (offering.institutionContext?.isBCPrivate &&
+        [
+          OfferingDeliveryOptions.Online,
+          OfferingDeliveryOptions.Blended,
+        ].includes(offering.offeringDelivered)) ||
+      !!offering.onlineInstructionMode,
+  )
+  // Allow the property only when the institution either BC Public or BC Private and the offering is online or blended.
+  @AllowIf(
+    (offering: OfferingValidationModel) =>
+      (offering.institutionContext?.isBCPrivate ||
+        offering.institutionContext?.isBCPublic) &&
+      [
+        OfferingDeliveryOptions.Online,
+        OfferingDeliveryOptions.Blended,
+      ].includes(offering.offeringDelivered),
+    undefined,
+    {
+      message: `${userFriendlyNames.onlineInstructionMode} is not allowed for provided institution type or offering delivery type and offering online delivery inputs.`,
+    },
+  )
+  @IsEnum(OnlineInstructionModeOptions, {
+    message: getEnumFormatMessage(
+      userFriendlyNames.onlineInstructionMode,
+      OnlineInstructionModeOptions,
+    ),
+  })
+  onlineInstructionMode?: OnlineInstructionModeOptions;
+
+  /**
+   * Specifies if the blended offering will always be provided with the same total duration of online delivery.
+   * Values can be "yes" or "no".
+   */
+  @ValidateIf(
+    // Validate when the property must have a value or it already has a value.
+    (offering: OfferingValidationModel) =>
+      (offering.institutionContext?.isBCPrivate &&
+        offering.offeringDelivered == OfferingDeliveryOptions.Blended) ||
```

Comment:
Is there a reason not to use triple comparison in this situation?

Reply:
It was missed in many. Nice catch! I added it